### PR TITLE
Michael's update

### PR DESCRIPTION
**Pull request by wadler**
Improved description of exercises in Preface, plus a few related changes. Suggested by Michael PJ.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,6 @@ permalink: /GettingStarted/
 <!-- Status & Version Badges -->
 
 [![Build Status][plfa-badge-status-svg]][plfa-badge-status-url]
-[![pre-commit.ci status][pre-commit-status-svg]][pre-commit-status-url]
```
Comment (wenkokke) [inline README.md:9]: Why did you remove this badge?

File: README.md [hunk 2]
```diff
@@ -248,34 +244,36 @@ You'll see the key sequence of the character in mini buffer.
 
 [Visual Studio Code][vscode] is a free source code editor developed by Microsoft. There is [a plugin for Agda support][vscode-agda] available on the Visual Studio Marketplace.
 
+### Atom
```
Comment (wenkokke) [inline README.md:247]: You seem to have added this section back in, but Atom is deprecated.

File: README.md [hunk 3]
```diff
@@ -248,34 +244,36 @@ You'll see the key sequence of the character in mini buffer.
...
+
+[Atom][atom] is a free source code editor developed by GitHub. There is [a plugin for Agda support][atom-agda] available on the Atom package manager.
+
 ## Getting Started for Contributors
 
 If you plan to build PLFA locally, please refer to [Contributing][plfa-contributing] for additional instructions.
 
 <!-- Links -->
 
-[plfa-badge-status-svg]: https://github.com/plfa/plfa.github.io/actions/workflows/build.yml/badge.svg
```
Comment (wenkokke) [inline README.md:257]: These changes also undo previous changes.

File: README.md [hunk 4]
```diff
@@ -248,34 +244,36 @@ You'll see the key sequence of the character in mini buffer.
...
+[agda-badge-version-url]: https://github.com/agda/agda/releases/tag/v2.6.2.2
+[agda-stdlib-version-svg]: https://img.shields.io/badge/agda--stdlib-v1.7.1-blue.svg
+[agda-stdlib-version-url]: https://github.com/agda/agda-stdlib/releases/tag/v1.7.1
 [plfa]: https://plfa.inf.ed.ac.uk
 [plfa-epub]: https://plfa.github.io/plfa.epub
 [plfa-contributing]: https://plfa.github.io/Contributing/
 [ghcup]: https://www.haskell.org/ghcup/
 [git]: https://git-scm.com/downloads
 [xcode]: https://developer.apple.com/xcode/
-[agda-readthedocs-installation]: https://agda.readthedocs.io/en/v2.6.3/getting-started/installation.html
```
Comment (wenkokke) [inline README.md:273]: These undo previous changes.

**Review by wenkokke — CHANGES_REQUESTED** [review]
Please don't merge this. This PR undoes a lot of previous changes.

**Comment by wadler** [comment]
Thanks. Good catch. I didn't go through the proper git steps the first time I made changes, and when I replayed the changes I inadvertently reverted some files. I've submitted a second pull request (tspl-update-michael-2) which I believe is now correct.

When I came in this morning, the checks on tspl-update-michael (the original) had not yet completed though they were running all night. That's a long time for checks! How long is it until they time out?

**Pull request closed without merging**